src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -365,6 +377,7 @@ protected override void Free()
             _analyzedNullabilityMapOpt = analyzedNullabilityMapOpt;
             _returnTypesOpt = returnTypesOpt;
             _snapshotBuilderOpt = snapshotBuilderOpt;
+            _defaultValues = new Dictionary<(SyntaxNode, ParameterSymbol), BoundExpression>();
```

Comment:
>new Dictionary<(SyntaxNode, ParameterSymbol), BoundExpression>() [](start = 29, length = 64)

Consider using `PooledDictionary<,>` and delaying the allocation until the first use, similar to other dictionary fields.